tests/test_base_loader.py

```diff
@@ -116,11 +121,21 @@ def test_replace_value(self):
         il.replace_value(None, 'Jim', lambda x: {'name': x})
         self.assertEqual(il.get_collected_values('name'), ['Jim'])
 
+    def test_replace_value_none(self):
+        il = CustomItemLoader()
+        il.replace_value('name', None)
+        self.assertEqual(il.get_collected_values('name'), [])
+        il.replace_value('name', 'marta')
+        self.assertEqual(il.get_collected_values('name'), ['Marta'])
+        il.replace_value('name', None)  # when replacing with `None` nothing should happen
+        self.assertEqual(il.get_collected_values('name'), ['Marta'])
```

Comment:
I'm ok to leave this as-is (it's not replaced by `None`) for historical reasons, but was this decision consciously?

Reply:
Looking at it through the test code, it looks like a bug. But I think there’s a rather high chance that there’s code relying on this behavior.

I’m +0.5 to deprecating this behavior (in a separate PR).